pkg/crds/crds.go

```diff
@@ -0,0 +1,205 @@
+// Package crds is used for installing rancher CRDs.
+package crds
+
+import (
+	"context"
+	"embed"
+	"fmt"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/rancher/rancher/pkg/fleet"
+	"github.com/rancher/wrangler/pkg/crd"
+	"github.com/rancher/wrangler/pkg/yaml"
+	apiextv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/selection"
+	"k8s.io/client-go/rest"
+)
+
+const (
+	// yamlDir directory holding CRDs yaml files.
+	yamlDir = "yamls"
+
+	// readyDuration time to wait for CRDs to be ready.
+	readyDuration = time.Minute * 1
+
+	k8sManagedByKey = "app.kubernetes.io/managed-by"
+	managerValue    = "rancher"
+)
+
+//go:embed yamls/*
+var crdFS embed.FS
+
+// EnsureRequired will ensure all required CRDs needed by Rancher based on the currently enabled features are installed and up to date.
+func EnsureRequired(ctx context.Context, cfg *rest.Config) error {
+	return Ensure(ctx, cfg, RequiredCRDs())
+}
+
+// Ensure will ensure all given CRDs are installed and up to date.
+func Ensure(ctx context.Context, cfg *rest.Config, crdNames []string) error {
```

Comment:
Nit: Near as I can tell, this isn't used except by `EnsureRequired` so we should probably make it not-exported.

Reply:
That use case makes sense to me. I think there's a couple odd things about this being exported, namely:
- It's currently not in use anywhere outside the package ( I think we can leave this one since we have an exact use case in mind)
- Another function that's also exported wraps it (which one should I use as a calling function? What makes a CRD required?)
- You need to know the `crdName` to request an install, but there's no indication on the syntax that should be used (i.e. should I use gvr/gvk? Lowercased? separated by "."?)

Reply:
Resolving in favor of adding a comment on how to identify the `crdNames`.